core/network.kt

```diff
@@ -0,0 +1,46 @@
+private fun isInternetConnected(context: Context): Boolean {
```

Comment:
Are you missing imports? Surprised this works?

Reply:
sorry sorry! It's just a draft. I need to clean things up and test

Reply:
Oh my b!

Reply:
you can put it in draft state if you want.

Reply:
ty tho! I'll add in the imports

Reply:
I'm also just excited to read some non-python code. Are you working in Android Studio? It can help you resolve these errors.

Reply:
:D ikrrr! I'm actually on vscode, but I'll make the switch! Kotlin is a beauty